walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,75 +21,111 @@
+        if (validEventTime.duration() >= requestDuration) {
+          times.add(validEventTime);
         }
+        eventStartTime = eventEndTime;
+        priorEventTime = currentEvent;
+      } else if (priorEventTime != null && priorEventTime.contains(currentEvent)) {
+        eventStartTime = priorEventTime.end();
+      } else {
+        eventStartTime = eventEndTime;
+        priorEventTime = currentEvent;
+      }
     }
     /* Add in event at end of day */
-    if(eventStartTime < 60*24 && 60*24 - requestDuration >= eventStartTime){
-        TimeRange validEventTime = TimeRange.fromStartEnd(eventStartTime,60*24, false);
-        times.add(validEventTime);
+    if (eventStartTime < 60 * 24 && 60 * 24 - requestDuration >= eventStartTime) {
+      TimeRange validEventTime = TimeRange.fromStartEnd(eventStartTime, 60 * 24, false);
+      times.add(validEventTime);
     }
     return times;
   }
 
-    /**
-        Returns sorted list of events that attendees are attending.
+  /**
+    * Returns list of events that have at least one attendee in it sorted by earliest time.
     */
-  public ArrayList<Event> sortEvents(ArrayList<Event> events, HashSet<String>attendees){
-    ArrayList<Event> sortedEvents = new ArrayList<Event>();
-    while(events.size()>0){
-        int minIndex = 0;
-        for(int i = 0; i < events.size(); i++){
-            if(events.get(i).getWhen().start() < events.get(minIndex).getWhen().start()){
-                minIndex=i;
-            }
-        }
-        Event earliestEvent = events.get(minIndex);
-        events.remove(minIndex);
-        HashSet<String> eventAttendees= new HashSet<String>(earliestEvent.getAttendees());
-        Boolean eventAttended = false;
-        for(String attendee : attendees){
-            if (eventAttendees.contains(attendee)){
-                eventAttended=true;
-            }
-        }
-        if(eventAttended){
-            sortedEvents.add(earliestEvent);
+  public ArrayList < Event > sortEvents(ArrayList < Event > events, HashSet < String > attendees) {
+    sort(events);
+    Boolean eventAttended = false;
+    ArrayList < Event > sortedEvents = new ArrayList < >(events);
+
+    for (Event event: events) {
+      HashSet < String > eventAttendees = new HashSet < String > (event.getAttendees());
+      eventAttended = false;
+      for (String attendee: attendees) {
+        if (eventAttendees.contains(attendee)) {
+          eventAttended = true;
         }
-        
+      }
+      if (eventAttended) {
+        sortedEvents.add(event);
+      }
+    }
+    return sortedEvents;
+  }
+
+  /**
+    * Splits a given ArrayList in half, sorts it, and returns the sorted ArrayList
+    */
+  public void merge(ArrayList < Event > events, int left, int middle, int right) {
+    ArrayList < Event > leftEvents = new ArrayList < >(events);
+    ArrayList < Event > righttEvents = new ArrayList < >(events);
```

Comment:
This formatting is weird — there should be no spaces around < for generics.

Also, `rightEvents` — this will produce a compile error.

Make sure you run tests too.

Reply:
Did you format this file with prettier instead of google-java-format?

Reply:
I used a third party formatter.